linux/dir_tools.bsh

```diff
@@ -470,4 +471,65 @@ function relative_path()
   else
     echo "${ans[*]}"
   fi
-}
\ No newline at end of file
+}
+
+#**
+# .. function:: strip_copy
+#
+# :Arguments: * ``$1`` - The source directory
+#             * ``$2`` - The destination directory
+#             * ``$3`` - Strip this many leading components from file names on ``cp``
+#
+# Replicates the behavior of ``tar --strip-components``, but when applied to a ``cp`` scenario.
+#
+# .. note:
+#
+#    ``cp`` will process files in a different order than ``tar --strip-components``. In case there are multiple files with the same name at a stripped depth,files may be overwritten differently than ``tar --strip-components`` would have. While ``tar`` keeps the first one in the archive order, ``cp`` will keep the last one in ``cp`` order. And as both order and behaviors are different, the result could be the same, or different.
+#**
+function strip_copy()
+{
+  local source_dir=$(real_path "${1}")
+  local destination=${2}
+  local depth=${3}
+
+  mkdir -p "${destination}"
+
+  # Technically the length of destination is not including in the xargs like
+  # formation internal the find command, however it turns out everyone's
+  # implementation of xargs/find adds such a large safety buffer that that
+  # discrepancy doesn't matter
```

Comment:
```suggestion
  # Technically, the length of the destination, DEST, is not included in the xargs
  # line formation because it is internal the find command, however it turns out
  # everyone's implementation of xargs/find adds such a large safety buffer that
  # the discrepancy doesn't matter
```